### DISCUSSION TITLE
Star trails not generating

### DISCUSSION THREAD

**Discussion by czimm74** [aaronwmorris/indi-allsky#1124] [General]
Not sure what is happening or not happening.  But the star trail image and time lapse have not been generating.  I have watched the logs while trying to run but I am not seeing any errors come up.  It also does not look like anything is miss configured.

**Comment by aaronwmorris** [comment]
Please check through the logs for exceptions.  If you updated recently, make sure you have re-run setup.sh to account for the database schema changes.

I am traveling for work this week, so it might take me longer to respond than normal.

**Comment by czimm74** [comment]
Had to delete and regenerate to find anything in the logs.  Looks like it excluded all the images, so it did not have enough images to generate the star trails.

It is not telling me in the log what it is excluding the 7500 images on though.

Target ADU night is 50
Deviation is 10

Under the StarTrails section I have this:
Sun -8
Exclude moon mode on
Max moon alt is -91
Phase is 101
StarTrails max ADU is 38
Max threshold is 40
Pixel threshold is 1
Timelapse min frames is 250

I did just an update today and reran setup to be sure it was all current.  I'll keep digging through and see if I can catch what is throwing off the startrails.

Only other thing I have noticed is that even though I have max exposure at 45 seconds, it hasn't been longer than 4 second exposures last few days.  Only thing I have tinkered with settings wise last week or so is dropping the ADU from ~ 85 down to 50.

**Reply by aaronwmorris** [reply]
Max threshold is very low, I recommend something like 190.  The `Max Threshold` and `Pixel Threshold` are linked settings which are if the Pixel threshold percentage of pixels exceeds max threshold (ADU), then the frame is excluded.

I know I need to create a wiki page on this topic.

**Reply by czimm74** [reply]
Yes, yes you do.  ;)

Ok, making that adjustment did let me generate a star trail image.  Still not sure why it stopped in the first place, it was before I started making small adjustments to the configs.

**Reply by aaronwmorris** [reply]
I have started a wiki page and linked to it from the config page.

https://github.com/aaronwmorris/indi-allsky/wiki/Startrails

**Comment by czimm74** [comment]
2024-02-07T07:25:42.534511-05:00 allsky-pi5 [INFO] Video-5-1556218/MainThread video.generateKeogramStarTrails() [859]: 
Processed 7400 of 7504 images
2024-02-07T07:26:00.145707-05:00 allsky-pi5 [INFO] Video-5-1556218/MainThread video.generateKeogramStarTrails() [859]: 
Processed 7500 of 7504 images
2024-02-07T07:26:01.047393-05:00 allsky-pi5 [ERROR] Video-5-1556218/MainThread video.generateKeogramStarTrails() [943]:
 Not enough frames to generate star trails timelapse: 0
2024-02-07T07:26:01.047523-05:00 allsky-pi5 [WARNING] Video-5-1556218/MainThread video.generateKeogramStarTrails() [948
]: Total keogram/star trail processing in 1128.3 s

**Reply by aaronwmorris** [reply]
I do not believe I have touched the algorithm for star trails lately.  Did you upgrade INDI?  If so, it is possible something has changed with regard to the driver that has altered the camera behavior.

**Reply by czimm74** [reply]
No, I have left INDI untouched in all my updates.  This is just a weird gremlin that is hiding somewhere.  Not ruling out a config change I made did something.
